### PR TITLE
Create google my business invite

### PR DESCRIPTION
**Pull request by cbauerman**
This PR adds the UX elements for the Google My Business nudge as we don't have any other screens to add the actions yet.

### Specs
See p989we-co-p2:

The Traffic tab in the Stats page will now show the following nudge to users identified in the signup survey:

Desktop
nudge

Mobile
mobile nudge

### Testing Instructions
- Boot this branch locally or use the calypso.live link below
- Create a site with a `promote` siteGoals if you don't have one. This can be done from the `http://calypso.localhost:3000/start/about` page by selecting at least the "Promote your business, skills, organization, or events" option.
- Navigate to this site stats page `/stats/day/:site_slug`
- Notice the default stat page
- Upgrade your site plan to business or premium
- Go back to the stats page
- Notice the GMB nudge
- Check that it displays properly for different types of screens and on various browsers.

### Reviews
- [ ] Code
- [ ] Product

### CODE REVIEW

File: client/components/google-my-business/stats-nudge.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+			<h2 className="google-my-business__stats-nudge-heading">
+				{ translate( 'Is your business listed on Google?' ) }
+			</h2>
+
+			<p className="google-my-business__stats-nudge-description">
+				{ translate( 'Google my business etc.' ) }
+			</p>
+
+			<Button className="google-my-business__stats-nudge-button">
+				{ translate( 'Drive More Traffic' ) }
```
Comment (a8ci18n) [inline client/components/google-my-business/stats-nudge.js]: Hi! I've found a possible matching string that has already been translated **2** times:
`translate( 'Drive traffic to your blog' )` *ES Score: 6*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/components/google-my-business/stats-nudge.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+					<h1 className="google-my-business__stats-nudge-heading">
+						{ translate( 'Is your business listed on Google?' ) }
+					</h1>
+					<h2 className="google-my-business__stats-nudge-description">
+						{ translate(
+							'Google My Business is a fast and free way to drive traffic to your local business.'
+						) }
+					</h2>
+					<div className="google-my-business__stats-nudge-button-row">
+						<Button primary>{ translate( 'Drive More Traffic' ) }</Button>
```
Comment (a8ci18n) [inline client/components/google-my-business/stats-nudge.js]: Hi! I've found a possible matching string that has already been translated **2** times:
`translate( 'Drive traffic to your blog' )` *ES Score: 6*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by Tug** [comment]
Great work here @cbauerman !

I think it's ready for review, we don't yet have the screens to handle the actions from that nudge so I say we keep it as a UX addition for now.

File: client/components/google-my-business/stats-nudge.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import Card from 'components/card';
+import SectionHeader from 'components/section-header';
+
+export const StatsNudge = ( { translate } ) => {
+	return (
+		<div>
+			<SectionHeader
+				className="google-my-business__stats-nudge-header"
+				label={ translate( 'We recommend Google My Business' ) }
+				href="#"
```
Comment (yurynix) [inline client/components/google-my-business/stats-nudge.js]: If it's not a link, why include `href` property?
https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-is-valid.md

File: client/components/google-my-business/stats-nudge.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+			/>
+			<Card className="google-my-business__stats-nudge">
+				<div className="google-my-business__stats-nudge-image-wrapper">
+					<img
+						className="google-my-business__stats-nudge-image"
+						src="/calypso/images/google-my-business/phone-screenshot.png"
+						alt="Your business with Google My Business"
```
Comment (yurynix) [inline client/components/google-my-business/stats-nudge.js]: IMO, we're missing `translate()` here

File: client/my-sites/stats/site.jsx [hunk 5]
```diff
@@ -223,10 +237,14 @@ export default connect(
 	state => {
 		const siteId = getSelectedSiteId( state );
 		const isJetpack = isJetpackSite( state, siteId );
+		const siteGoals = getSiteOption( state, siteId, 'site_goals' ) || '';
+		const sitePlan = getSitePlanSlug( state, siteId );
 		return {
 			isJetpack,
 			hasPodcasts: getSiteOption( state, siteId, 'podcasting_archive' ),
+			siteGoals: siteGoals.split( ',' ),
```
Comment (yurynix) [inline client/my-sites/stats/site.jsx]: Props always changing, `split()` creates new array each time

File: client/my-sites/stats/site.jsx [hunk 6]
```diff
@@ -132,6 +143,9 @@ class StatsSite extends Component {
 				/>
 				<div id="my-stats-content">
 					{ config.isEnabled( 'onboarding-checklist' ) && <ChecklistBanner siteId={ siteId } /> }
+					{ config.isEnabled( 'google-my-business' ) &&
```
Comment (yurynix) [inline client/my-sites/stats/site.jsx]: What you think about having that as a ([memoized](https://github.com/Automattic/wp-calypso/tree/master/client/lib/create-selector)) selector?

```js
const shouldShowGMB = ( state, siteId ) => {
   const siteGoals = ( getSiteOption( state, siteId, 'site_goals' ) || '' ).split( ',' );
   const sitePlan = getSitePlanSlug( state, siteId );
   return siteGoals.indexOf( 'promote' ) !== -1 &&
        [ PLAN_PREMIUM, PLAN_BUSINESS ].indexOf( sitePlan ) !== -1;
};
```

**Comment by yurynix** [comment]
Thank you for working on that @cbauerman, I've left a few minor comments.

File: client/components/google-my-business/stats-nudge.js (hunk 3 above)
Comment (Tug) [reply client/components/google-my-business/stats-nudge.js]: As you can see here https://github.com/Automattic/wp-calypso/blob/master/client/components/section-header/index.jsx#L19 SectionHeader accepts an `href` prop and if given will end up creating an anchor element in `Card`: https://github.com/Automattic/wp-calypso/blob/master/client/components/card/index.jsx#L56

Given that we'll want this to be clickable and probably link to the GMB settings page I find this a good workable solution to make it behave like a link even though it doesn't link to anything yet

Comment (cbauerman) [reply client/components/google-my-business/stats-nudge.js]: I will replace the `SectionHeader` href with an "X" to clarify that clicking it dismisses the card.

File: client/components/google-my-business/stats-nudge.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+// import Card from 'components/card';
+import DismissibleCard from 'blocks/dismissible-card';
+import SectionHeader from 'components/section-header';
+
+const StatsNudge = ( { translate } ) => {
+	return (
+		<DismissibleCard
+			className="google-my-business__stats-nudge"
+			preferenceName="show-google-my-business-nudge"
+			temporary
```
Comment (cbauerman) [inline client/components/google-my-business/stats-nudge.js]: The `temporary` property makes it so dismissing the nudge only lasts for the page session until we figure out how we really want it to be dismissed/presented.

File: client/components/google-my-business/stats-nudge.js [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+ * External dependencies
+ */
+import React from 'react';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal Dependencies
+ */
+import Button from 'components/button';
+// import Card from 'components/card';
```
Comment (yurynix) [inline client/components/google-my-business/stats-nudge.js]: Should we remove that?

File: client/components/google-my-business/stats-nudge.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+import DismissibleCard from 'blocks/dismissible-card';
+import SectionHeader from 'components/section-header';
+
+const StatsNudge = ( { translate } ) => {
```
Comment (yurynix) [inline client/components/google-my-business/stats-nudge.js]: Maybe a more specific name like `GMBStatsNudge` ( which is used later ) is better here?

**Comment by yurynix** [comment]
Looks good 👍 
Checked in chrome (and chrome mobile emulation) and FireFox.

**Comment by stephanethomas** [comment]
The image disappeared for me when I reduce the width of the window:

screenshot

Note this can be fixed in another iteration.

**Comment by Tug** [comment]
@stephanethomas That matches the last iteration on the design. The mockups shown in the PR description matches the first iteration.
👍 Merging now

**Pull request merged**